Mage.Sets/src/mage/cards/c/CaseOfThePilferedProof.java

```diff
@@ -0,0 +1,175 @@
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.game.permanent.token.ClueArtifactToken;
+import mage.game.permanent.token.Token;
+import mage.target.targetpointer.FixedTarget;
+
+/**
+ *
+ * @author DominionSpy
+ */
+public final class CaseOfThePilferedProof extends CardImpl {
+
+    public CaseOfThePilferedProof(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ENCHANTMENT}, "{1}{W}");
+
+        this.subtype.add(SubType.CASE);
+
+        // Whenever a Detective enters the battlefield under your control and whenever a Detective you control is turned face up, put a +1/+1 counter on it.
+        Ability initialAbility = new CaseOfThePilferedProofTriggeredAbility();
+        // To solve -- You control three or more Detectives.
+        Condition toSolveCondition = new PermanentsOnTheBattlefieldCondition(
+                new FilterCreaturePermanent(SubType.DETECTIVE, "You control three or more Detectives"),
+                ComparisonType.MORE_THAN, 2, true);
+        // Solved -- If one or more tokens would be created under your control, those tokens plus a Clue token are created instead.
+        Ability solvedAbility = new SimpleStaticAbility(new ConditionalReplacementEffect(
+                new CaseOfThePilferedProofReplacementEffect(), SolvedSourceCondition.SOLVED));
+
+        this.addAbility(new CaseAbility(initialAbility, toSolveCondition,solvedAbility)
+                .addHint(new CaseOfThePilferedProofHint(toSolveCondition)));
+    }
+
+    private CaseOfThePilferedProof(final CaseOfThePilferedProof card) {
+        super(card);
+    }
+
+    @Override
+    public CaseOfThePilferedProof copy() {
+        return new CaseOfThePilferedProof(this);
+    }
+}
+
+class CaseOfThePilferedProofTriggeredAbility extends TriggeredAbilityImpl {
+
+    CaseOfThePilferedProofTriggeredAbility() {
+        super(Zone.BATTLEFIELD, new AddCountersTargetEffect(CounterType.P1P1.createInstance()).setText("put a +1/+1 counter on it"));
+        this.setTriggerPhrase("Whenever a Detective enters the battlefield under your control and whenever a Detective you control is turned face up, ");
+    }
+
+    private CaseOfThePilferedProofTriggeredAbility(final CaseOfThePilferedProofTriggeredAbility ability) {
+        super(ability);
+    }
+
+    @Override
+    public CaseOfThePilferedProofTriggeredAbility copy() {
+        return new CaseOfThePilferedProofTriggeredAbility(this);
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        switch (event.getType()) {
+            case ENTERS_THE_BATTLEFIELD:
+            case TURNED_FACE_UP:
+                return true;
+        }
+        return false;
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        if (event.getType() == GameEvent.EventType.ENTERS_THE_BATTLEFIELD ||
+                event.getType() == GameEvent.EventType.TURNED_FACE_UP) {
```

Comment:
Just confirming that the sourceId and targetId work the same for both events?

Reply:
I should not be using sourceId for either I think, but targetId does work the same for both events, so I will switch to using exclusively that.